### ISSUE TITLE
Window positions in MDO area of Workspace weirdness (multiple monitors)

### ISSUE THREAD

**Issue by lrowen** [f4exb/sdrangel#1835]
So for a long time now I've had a workspace set up with several plugins loaded.  Just recently, SDRangel has stopped remembering the positions at close and has started up with the plugin windows far to the right, off the current visible section  of the workspace, and have to be manually moved back into place on every startup.  I made a short screencast video of the effect; my setup is a 4-monitor one, and this video is of the monitor on which I run SDRangel.  If you want a video of the full screen setup (a T-shape, three monitors across with the fourth monitor below the center top one) I can record that and link as well.  Here's a link to the screencast video:
[https://nebula.pari.edu/index.php/s/iFo9rftbDX2wDz5](https://nebula.pari.edu/index.php/s/iFo9rftbDX2wDz5)

As can be seen in the video, even after I save the configurations this occurs.  So I have to reconfigure the window positions every time I restart SDRangel now, which I didn't have to do prior to a few weeks ago.

Maybe I'm missing something, but I've not found a separate 'save Window positions' option.  

CHANGELOG shows version 7.15.4-1 but I track my fork in git; my git revision is:
```
ops@ops-sdr1:/opt/build/sdrangel$ git show
commit f6ea6f8a2e51c2a4edd3cee5e98808f6bff1ac04 (HEAD -> master, origin/master, origin/HEAD)
Merge: a0c75f040 8afa83c01
Author: Edouard Griffiths <f4exb06@gmail.com>
Date:   Sat Sep 16 10:30:40 2023 +0200

    Merge pull request #1830 from srcejon/fix_1827
    
    Remote TCP improvements

ops@ops-sdr1:/opt/build/sdrangel$ 

```
EDIT: I tried to upload the SDRangel.conf, even by adding .txt on the end, and github doesn't like it at all, probably due to the very long lines.  Let's see how that works in a CODE block....
```
[General]
ambe="@ByteArray(AAAABHicY2BgYAAAAAQAAQ==)"
audio="@ByteArray(AAABjXicrY5NCsIwEIW/qGsXnsCdIChJN15D6N4iNaJgqfRH6kbqDbyPh9NHyMKt4oT3JsPMvHkPjOnNCMV9YG5KQ2HDlhO1OKOkpeEceKmcc2SBjS8TXGCrrhMO7CjCTK0NTyWUqjyd6kqaCTx7XZloPmGlHRvhZmM+w7wU8b/+u6dfXMxJuUaVgql0PXt5auWtifVFl3Llr/Xfh+E+GA==)"
current="@ByteArray(AAAEJ3icfZK9b9NAGMaf80fd77htmqbfIVAoS2WnaatORMAGM0xIZ7UORHUcK3GE2IIKG0M2FoYiRkYGWBgQDPwViIW/ALGUovLcJUBUVZz1+D37nnvfn99zF0IkwtkPq0E7ShPDiRuFOKiHnmk9+fT962NLvAdwSuEXZUAP0QbMwxP1ZP3kbQE1xEjQRoo7nId4iJtoYA8Hgsuz1JeeRF35L6FFd8j1FE0E9Nb0m4jzRwM7xSHNXfTHCXVMhjHGV1xdPkYfyVBJs0zQZNIWlQ6WxykHw1M1Uc48Vxqo8wqIvc8d/3Ofhb1Bb6znDQL/22mO0/yMutuP6gNe/22UZszR+0BXjZkxOtuk51r9ujlU6QlYqa2/a8D7hum+/fFyvq2q9Crqfuio2oWO/aLLI4YwXL1kCqtiWR+OgIptXS8AckgI6VDD1Ag3jjKOMU4IG5PcmREr0uXzlChVpq1r6z8+Yoavs8KWswJJbsQv7Wx4vHx3zii+lXkBOU/DgijJRe5bglxmWAEbza/urFua66oOqjocqOLMaCb51dZ+M4jvh9FGK6gnUdhqtJt74Ua1FoW1OGmnybz62zqLV7r6x4VrENekLLLYAyxDRvGd6yBZWjo/YzONuJAsD3u94ZOws3pPN0v1Ca4lLGkLhQg4wsCwmTmCapEcpdgiOS5QmIA7aV5eO5IZvmCfkqkBiGkFMQOZJaBqVmHlfJgkaqcNhXOh6Hvl8s7mVtn3d7cVVrVcrW6WPG/T93w/LO3ubqGIzsWD3qmqM3UVK2ALg+eoWgl3RBMR0x0znZcFb9y8faviTog1TAIZzVmYgpwm0swAblbhzkLmiDtH5Sl1lJ/xGxsuxRo=)"
current-configuration="@ByteArray(AAATgHicrVdrcBtXFT5ndWWvZdlav+L4GcWJ47ws62EnsZvGKztxMLUbYTstTQusasmPRJYWSWbqtlA5LfyBtukzSQk0KZQ+eM9QUqCUEhgmbdNSKI+WGZjyoz9gmGGY0tJCwXxXq9iKbUFnYKXv3L3n3L3Ps985e4yYTS6ORCfCs7G0qRTHE+54eCY6b3OMHytgowjbzGhjKpIMxyejMc9ENJyeTUY9qXQ4mU6Gx49Ek/MTist63OnrnurqnvHv8gR2pEzF2ek9943Ozjav3+NrCdlUPXjy1PVnnjgfEuozY2dfnOc31pl2tX9uMj6bcgfNItKKmTVVaaugErbpDtGbITJKmUxnYyiRmk5PJ+LhmDuYSicT8cTMnHto+vpkODkXKlP1O4+RvELlqr6RrFuXqo/lbjVV/3TGuq1Q9bo3z9153ea3QpVqX/XTP2mA1qhiMqqBGgy2xjmKxbnHrNVptWLh93XjxlpY6kp8/p0eL34+rV5peVJrIK2RtCYKNau9Lweyl7aODDdzaL2qD599cGrDyT+GWlS9t95T/8g3bwltUPVR7ejhgcHx0EZVr7Uffcf5m8NGK/reRNTGu0ObVcpdoS2Lt8ZWZmMbc2b7IdQWcloFaKEURSmN3zTFaRK1fkrgLk3hrCZKSbRioJHGaQraJGyrt2mnjKePX33pebLJ7m3NEK1E4re4D62sW62KMyssHayYk+4VHjOZ8gf844k4Di8Wg9dM8Y+l1+gs+mK31emKCL5x/HbTZk+n50Z9JJTWjGGHUxVpA94er1seSe5ETLVyJJEOpxNJd/9id1qJdUwObGVp3jE55TGVkVZOukvICetatqAKokqFp6iKHVTNh/QaS72GFarFsGudPjzu8fm6PT5/N9UpFU9k6o0Vuz+e28kkyth/2P33ek4NlGnMOwPuhSix9ptvXllfOoNllia2UTMZ61g6IxnrgRbszIaG0UPD17SHDly9b8Tdf8CzfWhsr8fdtzfo3h/ySTc0NsHT2ph0v9i9u6lcD4jLX3+uRt8tgm5s3eVWsccqerOFOb1+JTmE09FYbDodvcgQh4sUedYhXuIAZZEDJDPk3k5TOPaHRt19g4MD7b55u/In6KRzVdN+CtEouamPBvEbgLP65GmtqvcWtBR+xl9glEABfWcBfVcB/Y4C+p0F9LsK6LtX0Q/mdmM1/WrrGiywrsEC6xossK7B3LqW7+hIwb22LKuNPVJgriMF5jRSYI9Gcnu0cuThAj5jWfwFLYGCls6Clq6CltX3S1p2ZYruyLHFp8DUkitM6qEO/MbBDzEglWWaMB0hD9gjCR7poCvpADoI0l7cW61mAMksKWgm0YcHmAJ6MewIWh/E4JdnLSkMm8iWm+BcB7ITCdIYrGnZk1lMpLIdwdhFDsmVpXyInGCWMo6Y5Y45XB3Dwx2RiOFiyasyXOsVWVagSrabVaoX1C3/ZrXqD/R0deMPoh16+im3sQZEUwtKWouwSmY9mZL6shxsNlWMXiSQxSjcbNE7CM1059H7eknvLaRtIGMj4mPrzXkMbfs/x0eieopAJ/WrtZDkSWZbld/r97V75X/s4vqNLcy0lcxtZG6n5YFWuQG9OxE+mzDnoyvrSyS/zNLBC5LrfKyE/EtpQ2DxFrOlKAaeQEybxLFNsaBpnOdhLqIjXEwxVmmGSyiOEJjgUkICRx/lMkpyOaVw5mnWaJYr6DzGeRZ4DngeuAC8ALwI/BR4CfgZ8HPgZeAXwC+BXwG/Bl4BXgUiOKFo3sKzVw1R0V+wnB+tUl9c+DLLeeb5Z9VThXJWr03Upg8aGUEZ++uyEbECn1SwbBvySiF+eBqOahd9Js6sCH6ItNNQgRIcsgNlKR4pw0aVo3Rxs6FBX8F+vVL0Np46hZRBQcpgt5LFPGesVVrOyjzRqMNz9ew3Glj6ldHElAvG5EY2vd4elNESvkkbiDYqbRq1SrFJijb0i2yPtmH47SjbAQ8ZHejVC/hQxanGKbM5+65ktmQLuQoqJrmITEn+DitWnmC7FrhjlfrSDl9qcVCmNL8fKR1oEQdaC9Stfi61OMksa17MD1LhGTMWTSVmk+NRjxmbTSdgMstbfN7Ozp2Brk5kWzu8Xm93NDox4Z/weru8gZ2+6I5oYIJcRBrYXmZm/62/uvfSXz1lGo5YviHXrtlI5oV2VuAMch9JK7Ftaj0t80mt1FZ8xu112oau0LUyvCflhOlIp3BXkFEJJ6jKc4JqyUg1lOM3hj9w1h/OIyt+dt3i1MenwvE4ymQ4Mp0IX/yimX9OhK1J2Vx9Oik2cc9rZMtKoYgTZFcwW5w1mNl0rEyMS6uh8rn3jyLTXkqN9WrRdz8cvkbonidH9DUi+LuJCb1W9D0O5VrR+0YgoNdZKV29uCxq9OkNIvg2nKoRU28CmoF1yNQ3i55j7a/RFtK3WsnyNqLtZPixYQGgE+gCdgA7gV1ANwLDbnHuyMkHkDzqt2KkPWLPSdR6Lb/VsdYgIQ4aA2i5X/TLib7PCiGDpL/fGucKVvQhEWQ8Nyz6QYJGCH1/ABgBRoEx4CBwFXq5WpxzYaAPij1DF14wrsEI+Gq6lozrsHcfIvowO4yPYEWRS5jRpk9bnweHrZ04YtVi2Zoxg9XHxbmeCy/oCXFuK2KUWdSfiMHx6AZ0NQdXuLGIP/mHV3f20k2Y7M3iGUkxHxe6LD5hFbdYXWYs27ylPGoVt1qR7zOlY9EZM5rMptHm7WTeUXrwysGxnrF9wyF3v3mnY3hfcDRb6zWOYbl3wfnuLr4qEUuHJ6PmPWTeS+Z9ZdlGVx0YGuvZ299r3I92x7EtJ6zP6dPYhTNED8GJHiHzUdIfy+q1b4mFCczgCdeI9Milj+zMtzcteaSmWK1sy1uBfOgsGU9iPt/Jexm+K1+G75H2FNH3KfN0/tfluxAeOPOfcX/Tynre1+UlFuMHTIKPAXcxC76bFcH3sE3wvSwE38d2wfdzkeDjXCz4BKuCT3KJ4AfYIfizXCr4FDsFf47LBH+eywU/yC7Bp1kTfIYrBD/ElYK/wFWCv8jVgh/mGsFf4jWCH+FawY/yWsGPcZ3gx7le8Je5QfBXuFHwV7lJ8Ne4WfDXeZ3gv/KC4DeleEuKv0nxthTvSPF3Kf4hxbtS/FOKf0mxAKGQFCyFIoVNCiGFHd0rRbxBKMW8XSgqbxZKiTQ4pCiVwilFmRTlUrik0DgklAoeEkolTwmlCqnb83UraCg+MROJziQi8xd4Xy5i1p3GN/nAjXBOm9h3GQ5CIE2w84sy1mjFYkGmWSrJmAmSZOSG1YaTKVPWuvi89JaFrLdcOTDs3iv7n43J73Z4i1m+XGm6akbnUunojDsX0FF+bHo8KonWrMhzqkrpVFWkVcuUAHQ20KiTTCW1tVwFqqVM/Y68LPB/+PoWD1M26hS9AmP1yvqSj15qkQFfMKJ3hNuz3irv8Kr+G1acneU=)"
current-featureset="@ByteArray(AAAIT3icpVVLbxtVFP7uZJw4zstO00BCH9OQgoDGr9BXJMBJUSpLGCynSxYzsceOydi+zIxLqxLJXrPpFqkb/gMVLBBCrAqkLW/okj/QTfeEb8bOQ8RXQmLufDNzv3PuOfeeO/ecuxBCipGKXbXaji+1kWbLaFoNGxWhSdvwKq7VrNlOsmpbftu1kzUvu5wtt5q+23Ic2+1WxdhdmsgJfe1vIKfpa/cAORTx/dsbGeja+Y4ZoYfhk+vplYyxYfm249R9+4ZrlbdtV0anSy3f8luuce3AZnxU3/trrmzGBOTYaCZ7OZlmy8THtYUv4xOITwJTQBxIANOa+BYngBkNn+dO6m/s7j7ErNDwHH0+fzgWc1rifmfeBLCH3qURCyijhSZ8uHw78HCt37dQ59smD4hQ02PPZwv4mkpT1s4di5i3v2a/t+bu1vCTIGRFEc2tfnpv87P7D4pa9JsbXzzqimdni0PR3CedcIZSj10vbhhr+fz6UqYb0Z6SO0PM4DqK2ICBNeTZ1rGEDPnZgXxaKVGPySq8qPhlBf+6gr+o4C8p+MsK/oqCvzqAz/ejNIgftK68Yl35/rr+HbmSMtolZbR7kkFeSopZlRRRLSmiUepH47jnguKv6UmySsmyUjI4KoX+fg+WXFJKrnSHhaRUJ97EVnj2JM/dClJsFWwiyZ5FtsmTGJzIJN8uv1JkJU9lijKHJzNFC+8fs/BR2JLUbfTtHI7fH1dGm4zL7+Ckp/i0Qt1NImCTZG8RIKOyX6ZeYM0Ls4yF7ZBr0WsK7+I9rnYVb/O7p9Xo+/LI1Khlh+va92Ie8/J/rAdWlqgrD33IESAqIhgVU4gFqXWMeRXjQmBCVORk7DavVKGQqlTMKYFcXH+LqSqX0FcNzm5aROSJaDq9Et5yJppdXrl4lTcT8ztff2WYs0KYYWIuLt4JasyOUfeMm3WvvunYRpU1YPFOpe1afr3V3DEa9Wbbt72kUbBuGbZj3wz5C9Q56OwYFbvm2lSSc6ePmGQJc1rMwe6+8aSch3wBnVN6mFdPJw7qkNEvRPEzvZpzljXHOFJzzgU1ZwHxF2EuCtE5//GRCjL03+sCcYr7shX+N75SZ57/tRXyAzXMlzi7l2G+wv14FfI1yAvgCNjsV1n0amIIW0JHnTv4gRjGthiBI6JoiFE0RQwtMQYpxvGhmIArJuFxl30RR1sk8EDs4Tvie+IHYpd4SDwiHhM/Ej8RPxO/EL8SvxG/E38QfxJPxN4/UcaLFg==)"
hwDeviceUserArgs=@ByteArray(AAAAC3icm8DAyNjAyMIABAAJ5gEY)
limeRFEUSBCalib=@ByteArray(AAAAC3icm8DAyNjAyMIABAAJ5gEY)
mainWindowGeometry="@ByteArray(AAAAQnicY7x54TQDMwMQsK8EEtwMDHyzGBhYOKF8AwQfqqoBXQYANfoHnQ==)"
mainWindowState="@ByteArray(AAAB13icfVCxEsFAEH0RNJRkjNoH+Ae+gToTDIPERIxRp1XofIRC4X90fkAZMefdOhyFvdm7vdu3b98tAAXanV6AWPkKuMdc30o3bk1MEGKBFRL0GA+xRhcRAkwdpuv0M4xpfAtLoofMJ4jhEzuRlxnjjVXppATvXpU5PaOGityUUhmMpIImrZEgJumSntjtNVTZ7RvMRJhz+ZQ9YMU/9K/YDrGhxBEFfyrdKsFbu7KIPtpmAO6RZ4Ant/7agW9p/lbvMTkWPSF7zX7Ht7d5PYyI8alhJT+2sCfSXfBl0tkxk5JTDxIPhiBOVA==)"
preferences=@ByteArray(AAAAfnicm8DAyFjAJFecUpSYl56ao1ecmFuQk1qcX1qUnKpXkFNakg+UKmBmKGBhYGBlcGBjceI589iBneXQ0n0lDBwMCZyMDAVcPHDtOfnpDNwMBTzsRmYKronFJQ68LC4xDAwMfAwM/AwJAoyMDIIMCUJAK4VZ3NxCwgEJGx+U)

[___gamepad_saved_states_v2]
id_12557=@Variant(\0\0\0\b\0\0\0\x2\0\0\0\xe\0\x62\0u\0t\0t\0o\0n\0s\0\0\0\b\0\0\0\x12\0\0\0\x6\0\x37\0\x30\0\x37\0\0\0\x2\0\0\0\r\0\0\0\x6\0\x37\0\x30\0\x36\0\0\0\x2\0\0\0\f\0\0\0\x6\0\x37\0\x30\0\x35\0\0\0\x2\0\0\0\xe\0\0\0\x6\0\x37\0\x30\0\x34\0\0\0\x2\0\0\0\xf\0\0\0\x6\0\x33\0\x31\0\x38\0\0\0\x2\0\0\0\v\0\0\0\x6\0\x33\0\x31\0\x37\0\0\0\x2\0\0\0\n\0\0\0\x6\0\x33\0\x31\0\x36\0\0\0\x2\0\0\0\x11\0\0\0\x6\0\x33\0\x31\0\x35\0\0\0\x2\0\0\0\t\0\0\0\x6\0\x33\0\x31\0\x34\0\0\0\x2\0\0\0\b\0\0\0\x6\0\x33\0\x31\0\x33\0\0\0\x2\0\0\0\a\0\0\0\x6\0\x33\0\x31\0\x32\0\0\0\x2\0\0\0\x6\0\0\0\x6\0\x33\0\x31\0\x31\0\0\0\x2\0\0\0\x5\0\0\0\x6\0\x33\0\x31\0\x30\0\0\0\x2\0\0\0\x4\0\0\0\x6\0\x33\0\x30\0\x38\0\0\0\x2\0\0\0\x3\0\0\0\x6\0\x33\0\x30\0\x37\0\0\0\x2\0\0\0\x2\0\0\0\x6\0\x33\0\x30\0\x35\0\0\0\x2\0\0\0\x1\0\0\0\x6\0\x33\0\x30\0\x34\0\0\0\x2\0\0\0\0\0\0\0\x6\0\x32\0\x38\0\x39\0\0\0\x2\0\0\0\n\0\0\0\b\0\x61\0x\0\x65\0s\0\0\0\b\0\0\0\b\0\0\0\x2\0\x35\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\0\0\0\x3\0\0\0\x2\0\x34\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\0\0\0\x3\0\0\0\x2\0\x33\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\0\0\0\x2\0\0\0\x2\0\x32\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\0\0\0\x2\0\0\0\x4\0\x31\0\x37\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\0\0\0\f\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\0\0\0\r\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\xff\xff\xff\xff\0\0\0\x4\0\x31\0\x36\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\0\0\0\xf\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\0\0\0\xe\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\xff\xff\xff\xff\0\0\0\x2\0\x31\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\0\0\0\x1\0\0\0\x2\0\x30\0\0\0\b\0\0\0\x3\0\0\0\x12\0m\0i\0n\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\x12\0m\0\x61\0x\0\x42\0u\0t\0t\0o\0n\0\0\0\x2\xff\xff\xff\xff\0\0\0\b\0\x61\0x\0i\0s\0\0\0\x2\0\0\0\0)

[configuration-1]
data=@ByteArray(AAATR3icrVhrcBvVFT5n98qS5df6lcTPKE7iOCSWJTkvhyRey4ldg50I2wmQQllhrR0TWVokmSGUFrkvflBIwjtpWhJKebU/2plSaKEUMrQTQqC09JV/aWf40Q5/OhTaUlr3u7uKrcRWy0y78nfu3nPu3se5Z7971keI2eKikWimNzllKTSjeq87YuvcMXM8Oh3PWIo7kfQlolMmxVi1zMZ0LBVNTJhx/7gZzUynTH86E01lUtGxg2ZqZlwRzuOEziKqR+85duLmU8+diQjPK6PPvzXD7y+3XLV9ga0hH8Y04/HJjDnqPGsVkeZm1jzKmkoqZlX3iu4skVGCvkobI8n0ZGYymYjGfT3pTCqZSE4d8g1O3pyKpg5Fyjz64SMkr0i5R19Fzm2FRx/N3Woe/Z6sc1vp0es+OH34hrYPI1WecM3LP2uA1qhmMmqAWgy2pHQEK/LlpqUtFbO/rxszlsFSVxwMbfYH8Atq9UrLC1oDaY2kNVGk2dP9Tqd9acvJ8DFHVnj0a9euNeSQLR79qu88+ae2o9+LrPTo4f7VbX+48VxklUeP/NQMfXT2LmM1+m4lWsPbIm0eyl2RtXO3xhXMxjrm7Pr9qM3mtArQQmkyKYPfJCVoArVeSuIuQ1FbY1IKrRhopDE6AG0KtsXbtFPWH+bzb79Bqt39rRDfJRIBIvWehXWnlTu7wNLBijXhWxAmE+lQZ2gsmcDmxeMIlQP8mgwVnUX4xX1364oI773mPUt1ZTKHRoIklNVZw4VIKtIQLQGf3JKLgeKpGk5moplkytc7151W7GyTF64sydumUrlNZaSVk14hZAjomrC9V0lUpfABqmYv1fB+vdZRL2GFlmLYZaVBPO4PBrv8wVAX1SmVz2XrjQXeH8t5MoUy/h+8/0n3qYGyjXl7wHc5I6mIbp5eWJ/fg8ssTaxSMxnLWQYjGSuAFnhmZcPI/qHr2yN7rt017Ovd418/OLrT7wvv7PH1R4IyDI1WYA2THhLbtjWV651i+7tna/VtoscH1213ih1O0W0X1uSKhYxw8dW+SAu3FClyryM8TwjKHCFImsi9nZbw9kdGfOGBgb724IxLeQ+6ZqCG+ilCI+SjMA3g14dgDcrdWlQfKGgp/EyowCidBfQbCug3FtBvKqDfXEC/pYC+axH9QM4bi+kXW9dAgXUNFFjXQIF1DeTWdblHhwv62rEsNvZwgbkOF5jTcAEfDed8tHDkoQIx41hCBS2dBS0bClo2FrQs7i9p2ZItui/HFl8BU0uusGgrdeA3Bn6IA2mbaaJ0kPxgjxR4pIN20x500EM7ce+0mgIks6ShmUAffuAA0I1hh9F6LwbfblvSGDZpl60Irj32RHpoFNaM7MlyE3nYhcO4grySK0t4P5WCWco4ZpV7D+HqGBrqiMWMCpa8Ko9rvdJmBapil1XtCYC65Z9V4wl1bt3YhT8Q7eDLL/mMJTjSloKSluFYJaueLEl9NgdbTZVzucHcKdzs0DsIzfLl0fsKSe8tpK0kYxXOx9V35jG0+n8+H4nqKQad1C/WwiZPa011KBAKtgfk3+jF9RtrmekKstaRtZ4uP2j5I+BezPcx4MWF9TySv9TSwbOS64KsRELzaUPn3C1mSyYGHseZNoFtO8CCJrGft3ARHWQ3xdlDU1xMCRyBSS4hi0vpVi6jFJdTGnueYY2muZLOYJzXgbPAG8A54E3gLeDnwNvAL4BfAu8AvwJ+DfwG+C3wO+A8EMMOmXkLt69aLA1nl/rxIvW5hV9mOcM887rntkKJakAVX3rt/QtZQVnXu7IRsYKYVLBsFXmlEK+eRKC6RNjCnhUhDpF2Gh6gGJvsRVmCR8rgqHKUFdxsaNBXckivEt2NJ04gZVCQMricZDEvGJcqLc/LPNGow3P1HDIaWMaV0cSUO4zJhxR6hatHnpaITVpJtEpZo9FqKVqlWIN+ke3ROgy/HmU74CejA70GgCCq2NUEZdvsdyW71i7kKshNchHZ4nwPFyFgnoXPbgJeWqQ+7+FLLV7KluT3I2MKbMBpoGWR+nw/l1pKySprnssP0tEpK26mk9OpMdNvxaczSZis8pZgYMOGzZ0bNyDb2hQIBLpMc3w8NB4IbAx0bg6am8zOcaog0sD2MjP7b/3VfZL+6inbcNCJDZk3aSrJvNDFCoJB+pG0YrV19UmZT2olqvuUL1CqDl6ta2V4T8oJ05FB4askowpBUJ0XBDWSkWopx2+MeGA7Hs4gK359+dzUxw5EEwmUqWhsMhm9+EUzc1Z0OZNSK8I6Kap44AKpthSKeJRcCmaLvQYzW96FiXFJDVRBX/8IMu351FivEeGHEfC1Qve/MKwvETsGz72pLxVhbLe+THQfO3Zcr3NSunpxpWmE9QYRxpeT0YipNwHNwHJk6m1i65H2C7SW9CucZHkd0XoyQnBYJ7AB2AhsAjYDW4AuHAzbxOmj4+MyefTYyeMOOWC3E7c61tpDOAeNPrTsF71yop9yjpAB0q9yxrmaFX1Q9DCeGxK9t+O5CPq+BhgGRoBRYC+wD71cK16NouV1Itx07LhxPUbAV9OnybgBvruR6DPsNW7CimKXMKOqT9pj6bc4njjo1OJ2zZjC6hPiNJyiJ8XpkWPHLauoNxlH4NHt6OoQQuGOIv7yH89v7qbPYrJ3ilckxXxO6LL4vFPc5XSZdWwzjvILTvFF5+T7asmoOWWZKTuNtu4l676SvbsHRreO7hqK+Hqtw96hXT0jdq3bOILlHkXw3e/el4xnohOm9QBZD5L1UJndaN+ewdGtO3u7jYfR7hG45VHnc/okvHCK6HEE0VNkPU36M7Ze+76YHccMnqsYlhE5/5Gd/UHrfERqitNKvbyVfOWfJ+MFzOeHeS/Dj+TL8CJpoJofU/blPEYR15Od2rt13N+9sD7PKJdajJ8wCT4CHGUWfD8rgh9gVfCDLAQ/xC7BD3OR4EfYLfhR9gg+xsWCj7NX8Ne4RPAJLhX8dS4T/A0uF/wYVwg+yZrgU1wp+HGuEvxNrhb8BNcI/hbXCn6Slwh+ipcKfpqXCX6G6wQ/y/WCv80Ngv/Cs4I/kOJDKf4qxd+k+LsUH0nxDyk+luKfUvxLilkIhaRgKRQpVCmEFC5uFkoRrxSKm9cLxcNtQimWBq8UJVKUSlEmRbkUFTxrvVG3gF4S41MxcyoZmznH/blt/PMTW/Cx3XcHok4Vu66UHsb57+K35CGiucXsrM3i8jCU7Iekr8YoZcqWtc53IONg1o6D3X1Dvp1yhOm4/CJHHFjllyutitqRQ+mMOeXLHdUob5scMyWFWpV54VIlw6WatBp52IOo+hp1kkmitoyrQaKUrd+Ul9/9D9/VLvlPHy9iajeM5Qvr89F3qUUe5YJxLse43Y5DeYeX8N/R4Zi4)
```

**Comment by srcejon** [comment]
I don't recall any recent changes to this code - it should save positions automatically. 

What happens if you close all windows and workspaces, exit SDRangel and restart, then recreate your workspace?

**Comment by lrowen** [comment]
I'll give that a try tomorrow as I set up for a pass.

**Comment by lrowen** [comment]
Ok, some solid data here.

Using my laptop, since I can reproduce the issue there, I shut it down, undocked, and started up in single monitor mode.  SDRangel restores the positions ok, although they're not in exactly the same position.

Redocking, I found the position the plugin windows go to to be dependent upon the monitor the SDRangel window is on.  If it's the leftmost monitor, the position restores correctly.  If it's the middle monitor, the position restores somewhat to the right.  If it's on the rightmost monitor, the position restores much farther to the right.
Single monitor screencast: [https://nebula.pari.edu/index.php/s/8HbbiBLSnffkskb](https://nebula.pari.edu/index.php/s/8HbbiBLSnffkskb)

Triple monitor screencast: [https://nebula.pari.edu/index.php/s/Dg78LcPzAzCptjB](https://nebula.pari.edu/index.php/s/Dg78LcPzAzCptjB)

In hindishgt, the issue first presented itself when I went from three to four monitors in OPS, with SDRangel on the middle monitor, which had been the leftmost monitor.  Aha.  Are the plugin windows' positions saved as offsets from the SDRangel workspace origin, or from the desktop origin?  And then restored as offsets from the workspace origin?

**Comment by lrowen** [comment]
NOTE: This appears to possibly be a QT5 bug in QtWidget::restoreGeometry when used in a multi-monitor configuration and MDI SubWindows.  It appears to be a rather old bug, and I don't see anywhere showing it's fixed.  The documentation for QWidget::restoreGeometry states it's used for 'toplevel' geometry, which is exactly the effect that I am seeing, and documented at:
[https://www.qtcentre.org/threads/39529-QMdiSubWindow-saveGeometry-restoreGeometry](https://www.qtcentre.org/threads/39529-QMdiSubWindow-saveGeometry-restoreGeometry)

There are some workarounds in several of the bug reports from way back, including one that completely reimplemnted restoreGeometry for MDI subwindows to base the coordinates off of the workspace and not the full desktop.

[https://bugreports.qt.io/browse/QTBUG-18648](https://bugreports.qt.io/browse/QTBUG-18648)
[https://bugreports.qt.io/browse/QTBUG-1645](https://bugreports.qt.io/browse/QTBUG-1645)

VERY OLD bug reports.

**Comment by srcejon** [comment]
Will try to find an extra monitor somewhere, so I can take a look.

**Comment by srcejon** [comment]
Also a problem on Windows, with both Qt 5 and 6.

As mentioned above, comment for QWidget::saveGeometry says it is for top-level widgets, and QMdiSubWindow isn't a top-level widget (as they have a parent).

Should be possible to roll our own version of saveGeometry anyway, as the code seems fairly simple:

For Qt 5.15 to 6.6, we have:

    quint16 majorVersion = 3;
    quint16 minorVersion = 0;
    const int screenNumber = QGuiApplication::screens().indexOf(screen());
    stream << magicNumber
           << majorVersion
           << minorVersion
           << frameGeometry()
           << normalGeometry()
           << qint32(screenNumber)
           << quint8(windowState() & Qt::WindowMaximized)
           << quint8(windowState() & Qt::WindowFullScreen)
           << qint32(screen()->geometry().width()) // added in 2.0
           << geometry(); // added in 3.0

**Comment by srcejon** [comment]
There's also a bug on Linux on single monitor setups, whereby windows aren't restored to their previous position anyway. They appear about 40 pixels lower than they should be.

**Comment by srcejon** [comment]
Should be fixed in 7.17.1.
